File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

/**
 * 品牌接口
 * @author Administrator
 *	
 */
public interface BrandService {
	/**
	 * 返回品牌列表
	 * @return
	 */
	List<TbBrand> findAll();
	
	/**分页查询品牌列表
	 * @param pageNum  当前页
	 * @param pagesize 每页记录数
	 * @return
	 */
	PageResult<TbBrand> findPage(TbBrand brand,int pageNum,int pagesize);
	
	/**
	 * 添加品牌
	 * @param brand
	 */
	void add(TbBrand brand);
	
	/**
	 * 根据ID查询品牌信息
	 * @param id
	 * @return
	 */
	TbBrand findOne(Long id);
	
	/**修改品牌信息
	 * @param brand
	 * @return
	 */
	void update(TbBrand brand);
	
	/** 根据id批量删除
	 * @param ids
	 */
	void delete(Long[] ids);
	
	
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
	// 查询所有品牌列表
	this.findAll = function(){
		return $http.get("../brand/findAll.do");
	}
	
	// 查询品牌分页列表
	this.findPage = function(page,size,entity){
		return $http.post("../brand/findPage.do?page=" + page + "&size=" + size,entity);
	}
	
	// 添加
	this.add = function(entity){
		return $http.post("../brand/add.do",entity);
	}
	
	// 修改
	this.update = function(entity){
		return $http.post("../brand/update.do",entity);
	}
	
	// 查询单个品牌
	this.findOne = function(id){
		return $http.get("../brand/findOne.do?id="+id);
	}
	
	// 删除
	this.dele = function(ids){
		if(ids.length > 0){
			return $http.get("../brand/delete.do?ids="+ids);
		}
	}
	
	// 查询品牌信息
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.do");
	}
	
});